consensus/istanbul/backend/handler.go

```diff
@@ -92,39 +73,16 @@ func (sb *Backend) HandleMsg(addr common.Address, msg p2p.Msg, peer consensus.Pe
 			return true, errors.New("No proxy or proxied validator found")
 		}
 
-		// Only use the recent messages and known messages cache for messages
-		// that are gossiped
-		if sb.isGossipedMsgCode(msg.Code) {
-			hash := istanbul.RLPHash(data)
-
-			// Mark peer's message
-			ms, ok := sb.peerRecentMessages.Get(addr)
-			var m *lru.ARCCache
-			if ok {
-				m, _ = ms.(*lru.ARCCache)
-			} else {
-				m, _ = lru.NewARC(inmemoryMessages)
-				sb.peerRecentMessages.Add(addr, m)
-			}
-			m.Add(hash, true)
-
-			// Mark self known message
-			if _, ok := sb.selfRecentMessages.Get(hash); ok {
-				return true, nil
-			}
-			sb.selfRecentMessages.Add(hash, true)
-		}
-
-		if msg.Code == istanbulConsensusMsg {
+		if msg.Code == istanbul.ConsensusMsg {
 			err := sb.handleConsensusMsg(peer, data)
 			return true, err
-		} else if msg.Code == istanbulFwdMsg {
+		} else if msg.Code == istanbul.FwdMsg {
 			err := sb.handleFwdMsg(peer, data)
 			return true, err
 		} else if announceHandlerFunc, ok := sb.istanbulAnnounceMsgHandlers[msg.Code]; ok { // Note that the valEnodeShare message is handled here as well
-			go announceHandlerFunc(peer, data)
+			go announceHandlerFunc(addr, peer, data)
```

Comment:
It seems like `addr` is the address derived from the peer's node public key. Is this intentional? It feels like the desired behavior, but want to confirm.

I think it could also be helpful to make this more clear in the code that the address isn't necessarily the signer address

Reply:
The main purpose of `addr` is for the gossip cache, as the one to mark message's sent/received from specific peers is keys on `addr`.

That's a good point in making it clear of what address it is, and I'll add comments for that.  This probably should be changed to use `nodeID` instead, but I was keeping what the original istanbul implementation did ....